lib/karafka/connection/queue_consumer.rb

```diff
@@ -83,6 +83,24 @@ def target
         close
       end
 
+      # @ retun the zookeeper hosts with the optional chroot inserted
+      def zookeeper_hosts
+        @zookeeper_hosts ||= begin
+          hosts = ::Karafka::App.config.zookeeper.hosts
+          chroot = zookeeper_chroot
+          hosts.map { |host| "#{host}#{chroot}" }
+        end
+        @zookeeper_hosts
+      end
+
+      # @return zookeeper chroot with leading and trailing slashes removed or nil when
+      #   it is not set
+      def zookeeper_chroot
+        chroot = ::Karafka::App.config.zookeeper.chroot
+        return '' unless chroot
+        '/' + chroot.split('/').reject(&:empty?).join('/')
```

Comment:
we will remove it, as there's a #76 that will ensure that we have a proper chroot provided :) (I will change - comment just as a reference)